pkg/trust/policy.go

```diff
@@ -54,14 +58,16 @@ type genericRepoMap map[string]json.RawMessage
 
 // DefaultPolicyPath returns a path to the default policy of the system.
 func DefaultPolicyPath(sys *types.SystemContext) string {
+	if sys != nil && sys.SignaturePolicyPath != "" {
+		return sys.SignaturePolicyPath
+	}
+	userPolicyFilePath := filepath.Join(homedir.Get(), userPolicyFile)
+	if _, err := os.Stat(userPolicyFilePath); err == nil {
```

Comment:
Should return error if err != nil && err != ENOEXIST.

Reply:
Mm, I thought the idea was that, if the system couldn't use the local `policy.json` (whether because it doesn't exist or something else), the idea was to try to use the system's `policy.json`, right? What kind of errors might happen when trying to open the file (besides it not existing)?

Don't get me wrong, I'm not trying to question you, just asking to educate myself

Reply:
I’d like to strongly argue that deviating from the c/image logic here is undesirable _in this PR_. The bug we are fixing is that `podman image trust` is not editing the same config file that `podman pull` is consuming, and any deviation will (re)introduce that kind of bug.

---

That said, failing here on other errors is plausible, and we do do that in quite a few other parts of the code base.

(Detailed explanation: In some cases it’s not clear what the right thing to do is (EPERM might mean that the user has a policy but that policy is being silently ignored because of wrong permissions, or it might mean that this is a very bare system and there really is not much of a home directory at all); in some cases like I/O errors failing is _very likely_ the right thing to do.)

I’m fine with making that change; but if we are going to do it, I think it _needs_ to be paired with an equivalent change in c/image. And, _also_, `c/common/pkg/config.DefaultConfig`. (compare https://github.com/containers/image/issues/1726 )

Reply:
Thank you very much for the explanation, I really appreciate it.

As I (unfortunately) cannot contribute much to this discussion, I leave it to you to find the best way to solve this problem and I will update the PR then.

Reply:
Well you could split the decision and just logrus.Warn if this happens. As Miloslav points out, it is not likely to happen but you could get EPERM or potentially other errors.

Reply:
Done, now a warning is logged if there is an error other than ENOEXIST